samples/ChatbotExamplev2/AdapterWithErrorHandler.cs

```diff
@@ -0,0 +1,31 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Microsoft.Bot.Builder.Integration.AspNet.Core;
+using Microsoft.Bot.Builder.TraceExtensions;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Logging;
+
+namespace ChatbotSample
+{
+    public class AdapterWithErrorHandler : BotFrameworkHttpAdapter
+    {
+        public AdapterWithErrorHandler(IConfiguration configuration, ILogger<BotFrameworkHttpAdapter> logger)
+            : base(configuration, logger)
+        {
+            OnTurnError = async (turnContext, exception) =>
+            {
+                // Log any leaked exception from the application.
+                logger.LogError(exception, $"[OnTurnError] unhandled error : {exception.InnerException}");
+
+                // Send a message to the user
+                await turnContext.SendActivityAsync("The bot encountered an error or bug.");
```

Comment:
nit: I'm not familiar with this API, but would it be better to only send one activity with a formated message string than sending 3 seperate activities?

Reply:
I believe the first one is for local logging, the second two are for logging messages to the bot client, and the third is for logging an exception trace to the bot client. so only the second two could be combined (which I can do if you prefer)

Reply:
I was focused on the 3 `SendActivityAsync` calls. Up to you if you want/can consolidate.

Reply:
there's only two, so I think i'll leave them as is